src/init.cpp

```diff
@@ -704,14 +704,14 @@ void ThreadImport(std::vector<boost::filesystem::path> vImportFiles)
             LogPrintf("Headers-only reindexing finished. Going on with blocks\n");
 
         nFile = 0;
-        uiInterface.InitMessage(_("Reindexing block from files..."));
         while (true)
         {
             CDiskBlockPos pos(nFile, 0);
             if (!boost::filesystem::exists(GetBlockPosFilename(pos, "blk")))
                 break; // No block files left to reindex
             FILE *file = OpenBlockFile(pos, true);
             if (!file) break; // This error is logged in OpenBlockFile
```

Comment:
This logic is broken.
If OpenBlockFile returns NULL (considering we've already done existence check - which only verifies the name of the file exists in file system but it's location has not been reached yet) it means we likely have an IO error. In such case the loop breaks as if it has traversed all the files and considers he task done which is not true.

Reply:
Do you think adding a retry and if failing again printing an error message on screen (at log level this is already the case) can be ok?
In first place I opted for inserting an assert but maybe this is a little too severe; indeed if there is some error when importing, the node would keep the blockchain up to current importing level and proceed downloading the reamining blocks from p2p network.
Do you have any other proposal?

Reply:
The problem here is that this thread is labelled "Importing" but actually serves the reindex purpose too.
If something goes south during reindex (or fast reindex) the data on disk is, as you say, to last importing level **but indexes** are inconsistent and we have wrongly stated that the reindex has completed correctly.
Which means all RPC calls which rely on correct indexes might return inconsistent data.

Reply:
I would say in this case an assert would be more appropriate

Reply:
Modified accordingly.